### DISCUSSION TITLE
How to update gradio to work with Image Browser Extension?

### DISCUSSION THREAD

**Discussion by edgardmello** [brkirch/stable-diffusion-webui#6] [Q&A]
The Image Browser Extension requires at least Gradio version 3.23.0.

**Comment by jrittvo** [comment]
Part of the point of this "easy install" version is that it does not update to untested pieces each time it loads.  Updates will be released as app updates, just like a typical app.  Updating Gradio to support the extension can potentially break other things.  If you can accept the risk and you know what you're doing, you can upgrade Gradio manually using pip with the venv activated or targeted.

**Reply by edgardmello** [reply]
I've already updated the gradio but it kept the same version as before. Tnks for the reply.

**Reply by jrittvo** [reply]
If it reports the same version after the upgrade as it did before, then it didn't actually upgrade, or at least not where the the app uses it from.

**Reply by edgardmello** [reply]
I see. That's why I asked here.

![CleanShot 2023-04-13 at 16 03 58](https://user-images.githubusercontent.com/859126/231858486-7a1f8aa5-3237-4ce9-8187-fb12c50e922f.jpg)

**Comment by jrittvo** [comment]
It needs to be updated in the `venv/lib/python3.10/site-packages` folder.  Does that one still have a `gradio-3.16.2.dist-info` folder in it or does it show 3.23.0?

**Reply by edgardmello** [reply]
Shows gradio-3.16.2.dist-info 

![CleanShot 2023-04-13 at 16 11 10](https://user-images.githubusercontent.com/859126/231859793-c1566821-9485-420b-99f7-0a7e1702f46d.jpg)

**Reply by jrittvo** [reply]
Yep.  You need to make the venv environment active before running the pip command, or add an argument to the pip command that gives it the path to the venv.  I'm sorry, but I use conda for my venvs, and I don't want to give you a wrong pip command.  Do a Google search for  "pip upgrade package in a venv" and you'll find help.

**Reply by edgardmello** [reply]
I used python to enable venv. I created a separate txt file with only the gradio==version to just update it.

I ended up solving the version issue. I put version 23.0 in the requirements.txt and requirements_versions.txt files

But now the css style is not working.

What I found interesting in this modified version are:

Sampler parameters Tab - negative guidance minimum sigma option and Token Merging Tab those ones I don't have in automatic1111 default installation.

**Reply by jrittvo** [reply]
Yeah.  That is the risk of messing with this app version.  You should be able to go back to the older version by editing the gradio==version, and doing that stuff over, but you might also need to delete (or rename to save a copy) config.json and/or style.css and/or ui-config.json if the run with the newer Gradio changed something in any of those that now will conflict with the older Gradio.  I'm not sure who writes what file or how forgiving the system is.  You can always delete the whole app folder and reinstall it.

**Reply by edgardmello** [reply]
I use --ui-settings-file and --ui-config-file outside webui folder. I'll go back to old default WebUI. But kudus to this project! Always nice seeing people helping the community.

**Reply by jrittvo** [reply]
Yes.  They haven't been added to the official version.  And tomesd (Token Merging) is a nice speed-up.  The  "minimum sigma" is a switch to keep Token Merging happy in some situations.  There is a pull request sitting at Automatic1111 for adding these that is about 2 weeks old.  Automatic1111 seems to batch updates every few weeks, rather than day by day.  It'll be added soon, I suspect.  You help by testing it in this app version.  But testing means bugs and crashes . . .

**Reply by edgardmello** [reply]
This version is really faster and less resources hungry.

**Reply by jrittvo** [reply]
If you are patient and careful (make lots of file backups) you can look through the recent pull requests at Automatic1111 and the merges at this git, and added the updated pieces manually to your regular Automatic1111.  I add one new thing each day or two.  Sometimes takes some fine tuning, but I am just a smart end user and not any sort of coder, and I have been able to find my way, and 'm learning a lot in the process.  My regular Automatic1111 currently is very similar to this app build, and yes, they are both faster and eat less memory.

**Reply by edgardmello** [reply]
I'm kind of in the same scenario as you are. I'm a designer with no code experience. A long time ago I was in the telecom industry working with some routers and switches. That's why I'm in tech stuff! I'm not so used to GitHub and it's why I don't make more experiments.

ChatGPT helps me a lot to achieve some things nowadays.

**Comment by jrittvo** [comment]
Ditto on ChatGPT.  Last week I decided to try this stuff at a very basic level, with `diffusers` from the Hugging Face git and python commands, in a venv.  The documentation at HF sucks.  Maybe they fixed some, or all, of it in the update to `diffusers` the past day or too.  Things like referring to pipeline.something.or.other when it is actually pipe.something.or.other, and ChatGPT has helped immensely with those things, and simple code questions.  I get good answers, well explained, as opposed to hit and miss with Google searches.  It is pretty neat to be able to use AI to learn about simple scripting for AI.
